File: Sushant_Agawane/divisor_game.cpp
class Solution {
public:
    bool divisorGame(int n) {
        int alice = 1, bob = 0;
        int i = 1;
        while(i <= n) {
            if(n % i == 0) {
                swap(alice, bob);
                n = n - i;
                i = 1;
            }else {
                i++;
            }
        }
        if(alice) return true;
        return false;
    }
};

File: Parth Mathur/divisorGame.java
class Solution {
  
 static boolean dp[] = new boolean[1001];
    public static boolean solve(int n, char person){
      if(n==1) return false;
      if(dp[n] != false) return dp[n];
      if(dp[n] == false){
        
      for(int i =1; i<n; i++){
        if(n%i == 0){
            if(person=='A'){
              boolean b = solve(n-i, 'B');
              if(!b) dp[n]= true;
            }else{
              boolean b = solve(n-i, 'A');
              if(!b) dp[n]=  true;
            }
        }
        
      }
      }
      return dp[n];
      
    }
    public boolean divisorGame(int n) {
 
      return solve(n,'A');
    }
}
